env/fs_posix.cc

```diff
@@ -1055,8 +1056,42 @@ static FactoryFunc<FileSystem> posix_filesystem_reg =
           f->reset(new PosixFileSystem());
           return f->get();
         });
-#endif
 
+static FactoryFunc<FileSystem> zenfs_filesystem_reg =
+    ObjectLibrary::Default()->Register<FileSystem>(
+        "zenfs://.*", [](const std::string& uri, std::unique_ptr<FileSystem>* f,
+                         std::string* errmsg) {
+          std::string devID = uri;
+          FileSystem* fs = nullptr;
+          Status s;
+
+          devID.replace(0, strlen("zenfs://"), "");
+          if (devID.rfind("dev:") == 0) {
+            devID.replace(0, strlen("dev:"), "");
+            s = NewZenFS(&fs, devID);
```

Comment:
It is not needed, as the dev: mode it directly names a zoned block device.  I think it is useful to keep it like this, in case of ListZenFileSystems() failing for some reason.